it/test/controllers/CalculationPollingControllerISpec.scala

```diff
@@ -16,146 +16,245 @@
+                  LiabilityCalculationError(INTERNAL_SERVER_ERROR, "error"))
+                res.futureValue should have(
+                  httpStatus(INTERNAL_SERVER_ERROR)
+                )
+                IncomeTaxCalculationStub.verifyGetCalculationResponseByCalcId(testNino, "idFive", testTaxYear, noOfCalls = 6)
+              }
+            }
+          }
         }
+        testAuthFailures(path, mtdUserRole)
 
-        "has the status of SEE_OTHER (303)" in {
-          result.status shouldBe SEE_OTHER
-        }
+      }
+    }
 
-        s"redirect to '${controllers.routes.FinalTaxCalculationController.show(testTaxYear).url}''" in {
-          result.header("Location").head shouldBe controllers.routes.FinalTaxCalculationController.show(testTaxYear).url
+    s"GET $finalCalcPath" when {
+      s"a user is a $mtdUserRole" that {
+        "is authenticated, with a valid enrolment" should {
+          if (mtdUserRole == MTDSupportingAgent) {
+            testSupportingAgentAccessDenied(path, additionalCookies)
+          } else {
+            "redirects to final tax calculation page" when {
+              "the session key contains calculationId" in {
+                stubAuthorised(mtdUserRole)
+                IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesAndUkProperty)
+                IncomeTaxCalculationStub.stubGetCalculationResponseByCalcId(testNino, testTaxYear, "idOne")(
+                  status = OK,
+                  body = liabilityCalculationModelSuccessful
+                )
+
+                val result = buildGETMTDClient(finalCalcPath, additionalCookies ++ Map(SessionKeys.calculationId -> "idOne")).futureValue
+                val expectedRedirectUrl = if(isAgent) {
+                  routes.FinalTaxCalculationController.showAgent(testYearInt).url
+                } else {
+                  routes.FinalTaxCalculationController.show(testYearInt).url
+                }
+                result should have(
+                  httpStatus(SEE_OTHER),
+                  redirectURI(expectedRedirectUrl)
+                )
+              }
+
+              "calculation service returns retryable response back initially and then returns success response before interval time completed" in {
+                stubAuthorised(mtdUserRole)
+                IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesAndUkProperty)
+                IncomeTaxCalculationStub.stubGetCalculationErrorResponseByCalcId(testNino, "idFour", testTaxYear)(NO_CONTENT,
+                  LiabilityCalculationError(NO_CONTENT, "no content"))
+
+                val res = buildGETMTDClient(finalCalcPath, additionalCookies ++ Map(SessionKeys.calculationId -> "idFour"))
+
+                //After 1.75 seconds responding with success message
+                Thread.sleep(1750)
```

Comment:
Please don't use Thread.sleep anywhere in our code base

Reply:
This is a preexisting tests and not something I introduced. In this case as well they are testing polling so the thread.sleep is delaying the successful stub call not waiting for a result so the thread.sleep approach is not wrong in this scenerio.

Reply:
Next story created to address this issue:
https://jira.tools.tax.service.gov.uk/browse/MISUV-8977